File: GreenSignal/Data/Models/PetitionAttribute.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Data.Models
{
    public class PetitionAttribute
    {
        [Key]
        public string Id { get; set; }

        [Required]
        public Guid PetitionId { get; set; }

        public Petition Petition { get; set; }

        [Required]
        public string Name { get; set; }

        public string? StringValue { get; set; }

        public double? NumberValue { get; set; }

        public bool? BoolValue { get; set; }
    }
}

File: GreenSignal.Mobile/app/src/main/java/com/example/greensignal/domain/model/response/PetitionAttribute.kt
package com.example.greensignal.domain.model.response

import com.example.greensignal.data.remote.dto.response.PetitionAttributeDto

data class PetitionAttribute(
    val id: String = "",
    val name: String = "",
    val stringValue: String? = null,
    val numberValue: Double? = null,
    val boolValue: Boolean? = null
)

fun PetitionAttributeDto.toPetitionAttribute(): PetitionAttribute{
    return PetitionAttribute(
        id = id,
        name = name,
        stringValue = stringValue,
        numberValue = numberValue,
        boolValue = boolValue
    )
}